docs-src/builder/upcloud.mdx

```diff
@@ -0,0 +1,64 @@
+---
+description: >
+  This is a builder plugin for Packer which can be used to generate storage templates on UpCloud. 
+  It utilises the UpCloud Go API to interface with the UpCloud API.
+page_title: UpCloud - Builders
+---
+
+
+# UpCloud
+
+Type: `upcloud`
+
+The upcloud builder is used to generate storage templates on UpCloud.
+
+<!-- Builder Configuration Fields -->
+
+### Required
+
+@include 'builder/upcloud/Config-required.mdx'
+
+<!--
+  Optional Configuration Fields
+
+  Configuration options that are not required or have reasonable defaults
+  should be listed under the optionals section. Defaults values should be
+  noted in the description of the field
+-->
+
+### Optional
+
+@include 'builder/upcloud/Config-not-required.mdx'
+
+#### IP Address object (IPAddress)
```

Comment:
Should we put this below `Network Interfaces object` section, as it is used only there?

Reply:
Good point. I'll move it.